libraries/chain/controller.cpp

```diff
@@ -3476,11 +3476,14 @@ std::optional<chain_id_type> controller::extract_chain_id_from_db( const path& s
 
       if( db.revision() < 1 ) return {};
 
-      return db.get<global_property_object>().chain_id;
-   } catch( const bad_database_version_exception& ) {
-      throw;
-   } catch( ... ) {
-   }
+      auto * gpo = db.find<global_property_object>();
+      if (gpo==nullptr) return {};
+
+      return gpo->chain_id;
+
+   } catch (std::system_error &) {   
+   
```

Comment:
Can you add a comment about why we are ignoring `std::system_error`. Also please follow code style found in file.

Reply:
> Can you add a comment about why we are ignoring `std::system_error`. Also please follow code style found in file.

Sure, this was done to handle situation when there was no database and it needs to be created (db_error_code::not_found" exception). Not sure if the right way to plug it here or fix it differently.

Reply:
Sorry, I meant to add a comment to the code.

Reply:
Ok, got it :)